tests/core/runners/test_ase.py

```diff
@@ -47,6 +52,196 @@ def teardown_function():
             os.remove(os.path.join(SETTINGS.RESULTS_DIR, f))
 
 
+@pytest.fixture()
+def setup_test_environment(tmp_path):
+    reactant = Atoms(
+        symbols="CCHHCHH",
+        positions=[
+            [1.4835950817281542, -1.0145410211301968, -0.13209027203235943],
+            [0.8409564131524673, 0.018549610257914483, -0.07338809662321308],
+            [-0.6399757891931867, 0.01763740851518944, 0.0581573443268891],
+            [-1.0005576455546672, 1.0430257532387608, 0.22197240310602892],
+            [1.402180736662139, 0.944112416574632, -0.12179540364365492],
+            [-1.1216961389434357, -0.3883639833876232, -0.8769102842015071],
+            [-0.9645026578514683, -0.6204201840686793, 0.9240543090678239],
+        ],
+    )
+
+    product = Atoms(
+        symbols="CCHHCHH",
+        positions=[
+            [1.348003553501624, 0.4819311116778978, 0.2752537177143993],
+            [0.2386618286631742, -0.3433222966734429, 0.37705518940917926],
+            [-0.9741307940518336, 0.07686022294949588, 0.08710778043683955],
+            [-1.8314843503320921, -0.5547344604780035, 0.1639037492534953],
+            [0.3801391040059668, -1.3793340533058087, 0.71035902765307],
+            [1.9296265384257907, 0.622088341468767, 1.0901733942191298],
+            [-1.090815880212625, 1.0965111343610956, -0.23791518420660265],
+        ],
+    )
+    return reactant, product
+
+
+@pytest.mark.parametrize(
+    (
+        "nimages",
+        "convergence_tolerance",
+        "max_iterations",
+        "max_micro_iterations",
+        "morse_scaling",
+        "geometry_friction",
+        "distance_cutoff",
+        "save_raw_path",
+        "expected_length",
+    ),
+    [
+        (20, 2e-3, 15, 20, 1.7, 1e-2, 3.0, None, 20),  # Default parameters
+        (10, 2e-3, 15, 20, 1.7, 1e-2, 3.0, None, 10),  # Different number of images
+        (
+            20,
+            1e-4,
+            10,
+            10,
+            1.5,
+            0.01,
+            2.5,
+            "raw_path.xyz",
+            20,
+        ),  # Different interpolation parameters and save_raw
+    ],
+)
```

Comment:
Here and throughout the entire test suite changes, please do not use `pytest.mark.parameterize`. This is not the appropriate thing to do in your tests because it makes it extremely hard to tell at a quick glance what each parameter is doing. `@pytest.mark.parameterize` is for running a parameter sweep where you want to run the same test but with multiple values for a single parameter. It's not meant to run a single test. Please define the test function with no decorator and no arguments. Simply put the test values inside the function definition wherever they belong.